app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -0,0 +1,226 @@
+package ch.hikemate.app.model.route
+
+import android.util.JsonReader
+import java.io.Reader
+import okhttp3.OkHttpClient
+import okhttp3.Request
+
+/**
+ * Overpass implementation of the hiking route provider repository.
+ *
+ * This implementation uses the Overpass API to fetch the hiking routes from OSM.
+ *
+ * @see <a href="https://dev.overpass-api.de/overpass-doc/">Overpass API documentation</a>
+ */
+class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesRepository {
+
+  /** The URL of the Overpass API interpreter. */
+  private val OVERPASS_API_URL: String = "https://overpass-api.de/api/interpreter"
+  /** The type of format to request from the Overpass API, written in OverpassQL. */
+  private val JSON_OVERPASS_FORMAT_TAG = "[out:json]"
```

Comment:
Nitpick: These two could be constant.

Reply:
Mmmmmh, indeed, but wasn't thinking that setting a top level const could be done privately! Corrected thanks!